components/dialog/stories/template.js

```diff
@@ -24,65 +24,86 @@ export const Template = ({
 	footer = [],
 	customClasses = [],
 	id = getRandomId("dialog"),
+	size = "medium",
+	layout,
+	hasHeroImage = false,
+	heroImageUrl,
+	customStyles = {},
 } = {}, context = {}) => {
-	const { globals = {}, updateArgs } = context;
-	const scale = globals.scale ?? "medium";
-	const toggleOpen = function () {
-		updateArgs({ isOpen: !isOpen });
-	};
+	const { updateArgs } = context;
+	const toggleOpen = () => updateArgs({ isOpen: !isOpen });
 
 	const Dialog = html`
 		<div
 			class=${classMap({
 				[rootClass]: true,
-				[`${rootClass}--${scale}`]: true,
-				[`${rootClass}--dismissable`]: isDismissable && !isFullScreen && !isFullScreenTakeover,
-				[`${rootClass}--fullscreen`]: isFullScreen,
-				[`${rootClass}--fullscreenTakeover`]: isFullScreenTakeover,
+				[`${rootClass}--dismissable`]: isDismissible && (layout !== "fullscreen" || layout !== "fullscreenTakeover"),
+				[`${rootClass}--${layout}`]: typeof layout !== "undefined" && layout !== "default",
+				[`${rootClass}--${size}`]: typeof size !== "undefined", 
+				[`${rootClass}--noDivider`]: !hasDivider,
 				...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 			})}
 			id=${ifDefined(id)}
 			role="dialog"
 			tabindex="-1"
 			aria-modal="true"
+			style=${ifDefined(styleMap(customStyles))}
 		>
 			<div class="${rootClass}-grid">
-				${when(heading, () => [
-					html`<h1 class="${rootClass}-heading">${heading}</h1>`,
+				${when(hasHeroImage, () =>
+					html`
+						<div 
+							class="spectrum-Dialog-hero"
+							style="background-image:url(${heroImageUrl ?  heroImageUrl : "example-card-portrait.png"})">
```

Comment:
```suggestion
							style="background-image:url(${heroImageUrl})">
```

Can simplify this if we use the heroImageUrl's presence as the boolean

Reply:
https://github.com/adobe/spectrum-css/pull/2833/commits/340ef9b9c38362f75fc46a01f92432f98d9523b8